catkin_ws/src/state_estimation/src/hydrophones_bearing.py

```
@@ -1,269 +1,116 @@
-            PingerBearing_msg.pinger1_bearing.y = bearing_vector_global[1]
-            PingerBearing_msg.pinger1_bearing.z = 0
+def solve_bearing_vector(distance, is_three_hydrophones):
+    position = np.array([hydrophones_dx, hydrophones_dy, 0] if is_three_hydrophones else [hydrophones_dx, hydrophones_dy, hydrophones_dz])
+    bearing_vector = np.where(position != 0, distance / position, 0)
+    return bearing_vector
 
-        if msg.is_pinger2_active:
-            dt_hx = msg.dt_pinger2[0]
-            dt_hy = msg.dt_pinger2[1]
-            measurements = calculate_time_measurements(dt_hx, dt_hy)
-            bearing_vector_local = solve_bearing_vector(
-                measurements[0], measurements[1]
-            )
-            bearing_vector_global = quaternion.rotate_vectors(
-                auv_rotation,
-                np.array([bearing_vector_local[0], bearing_vector_local[1], 0]),
-            )
-            PingerBearing_msg.pinger2_bearing.x = bearing_vector_global[0]
-            PingerBearing_msg.pinger2_bearing.y = bearing_vector_global[1]
-            PingerBearing_msg.pinger2_bearing.z = 0
 
-        if msg.is_pinger3_active:
-            dt_hx = msg.dt_pinger3[0]
-            dt_hy = msg.dt_pinger3[1]
-            measurements = calculate_time_measurements(dt_hx, dt_hy)
-            bearing_vector_local = solve_bearing_vector(
-                measurements[0], measurements[1]
-            )
-            bearing_vector_global = quaternion.rotate_vectors(
-                auv_rotation,
-                np.array([bearing_vector_local[0], bearing_vector_local[1], 0]),
-            )
-            PingerBearing_msg.pinger3_bearing.x = bearing_vector_global[0]
-            PingerBearing_msg.pinger3_bearing.y = bearing_vector_global[1]
-            PingerBearing_msg.pinger3_bearing.z = 0
+def cb_hydrophones_time_difference(msg):
+    if not is_active or msg.frequency not in frequency_types:
```

dont need to check if frequency is valid, should just publish all frequencies (they'll always be one of the ones we define as possible)

---

although could be a useful sanity check, but if its a sanity check it needs to be verbose and not silently fail

---

It's just to make sure that if electrical fucks it up, we don't get fucked as well

---

sensor status checks if the frequency received is one of the correct frequencies, but I was worried in case there is some delay and the is_active isn't updated in time to stop the bearing  calculations

---

it doesn't have to be verbose because we will see the display turning red

---

yeah no i think ur right to check, just add a print/error log statement or something so it doesnt fail silently. but yeah i agree with you actually it makes sense to double check

---

github comments smh

---

we should still have a print statement though, there's 100 reasons the display could turn red and in general we shouldn't rely entirely on the display, its just useful as a quicker debugging tool